### ISSUE TITLE
Override toString() in Table class

### ISSUE THREAD

**Issue by sguptatw** [getgauge/gauge-java#53]
### Expected behavior
To string of the Table parameter should give all the contents of the table

### Actual behavior
To string of the Table parameter is the default implementation

### Steps to reproduce
1. Create a spec
```
Something
=========
wwwwwwww
--------
* Verify email text is <table:/email.csv>
```
2. add email.csv
```
Id,Name
1,The Way to Go On
2,Ivo Jay Balbaert
```
3. Give the implementation 
```
    @Step("Verify email text is <table:/email.csv>")
    public void implementation7(Object arg0) {
        System.out.println(arg0.toString());
    }
```
4. The to String implementation should give all the contents of the table 
### Gauge version
```
Gauge version: 0.4.1.nightly-2016-04-19

Plugins
-------
html-report (2.1.0)
java (0.3.4.nightly-2016-04-19)
```

**Comment by maquessime** [comment]
@sguptatw @mahendrakariya not able to reproduce here, I always get an empty string and not the default toString of Object.toString

Moreover, the type of Object is already String when I try to reproduce.

However, if we want the content of the table, should we force Table for arg0 instead of Object? or we want to support someone who put Object instead of Table?

Many thanks 😄

**Comment by mahendrakariya** [comment]
@maquessime We want to support both `Object` as well as `Table` for a table param.

**Comment by maquessime** [comment]
@mahendrakariya ok. If we have

`@Step("Verify email text is <table:/email.csv>")
    public void implementation7(Object arg0) {
        System.out.println(arg0.getClass().getName().toString());
    }`

Do we expect to have String or Table to be output?

**Comment by mahendrakariya** [comment]
@maquessime In the use case that you mentioned, we would expect the param type to be `Table`.

But let's consider a slightly more complicated use case. Let's have one concept and a few specs.
The concept can be something like this.
```
# Concept with <arg>
* Some random step with <arg>
```

The user can call this concept in the spec by either passing a table or a static / dynamic param.
```
* Concept with "static argument"
```
```
* Concept with 
     | Header 1 | Header 2 |
     |----------|----------|
     | cell 1   | cell 2   |
```

While implementing this step, the param type can be a `String` or a `Table`. In such cases, we would like the users to use `Object` type.
```Java
@Step("Concept with <arg>")
public void implementation(Object arg) {
    System.out.println(arg);
}
```

**Comment by apoorvam** [comment]
This is fixed by @maquessime in this [PR](https://github.com/getgauge/gauge-java/pull/56). Should be available in nightly >= 14.06.2016.
